test/integration/lib/nupnp_PhilipsHue.js

```diff
@@ -3,19 +3,25 @@
 var express = require ('express');
 
 var philips_nupnp_page = (function() {
-  return {
-    start: function(hue_id,hue_ipaddress,port) {
-      var app = express();
+    var _app = express();
+    var instance;
 
-      app.get('/', function (req, res) {
+    function start(hue_id,hue_ipaddress,port) {
+
+      _app.get('/', function (req, res) {
         res.send([{'id':hue_id,'internalipaddress':hue_ipaddress}]);
       });
       
-      app.listen(port, function () {
+      instance = _app.listen(port, function () {
         console.log('Philips nupnp app listening on port ' + port);
       });
     }
-  };
+
+    function stop() {
+      instance.close();
```

Comment:
`close()` is [an async method](https://nodejs.org/api/http.html#http_server_close_callback).